scripts/db_query_handler.py

```diff
@@ -0,0 +1,158 @@
+"""
+This module defines the DBQueryHandler class for handling database queries.
+"""
+
+import os
+import sqlite3
+
+from scripts.utils.files import (
+    is_readable_existing_file,
+    is_writeable_file_directory,
+    is_path_existent,
+)
+from scripts.utils.config import get_key_str
+
+
+class DBQueryHandler:
+    """
+    A handler for database queries.
+
+    Attributes:
+        schema_file (str): Path to the schema file.
+        db_path (str): Path to the SQLite database file.
+        delete_db (bool): Flag indicating whether to delete the existing database.
+        conn (sqlite3.Connection): SQLite database connection.
+        cursor (sqlite3.Cursor): SQLite database cursor.
+    """
+
+    def __init__(self, config: dict, delete_db: bool = False) -> None:
+        """
+        Initializes the DBQueryHandler with the given configuration and deletion flag.
+
+        Parameters:
+            config (dict): Configuration dictionary.
+            delete_db (bool): Deletion flag.
+        """
+        self.schema_file = get_key_str(config, "schema")
+        self.db_path = get_key_str(config, "db")
+        self.test_mode = get_key_str(config, "mode") == "test"
+        self.delete_db = delete_db
+        self._handle_database_deletion()
+        self._connect_to_database()
+        self._create_database_if_not_exist()
+
+    def _handle_database_deletion(self) -> None:
+        """
+        Deletes the existing database file if needed and if the file is readable.
+        """
+        if self._is_need_to_delete() and is_readable_existing_file(self.db_path):
+            os.remove(self.db_path)
+
+    def _is_need_to_delete(self) -> bool:
+        """
+        Checks if the database should be deleted due to delete_db or test_mode being True.
+
+        Returns:
+            bool: True if deletion is needed, False otherwise.
+        """
+        return self.delete_db or self.test_mode
+
+    def _connect_to_database(self) -> None:
+        """
+        Connects to the SQLite database.
+        """
+        if is_writeable_file_directory(self.db_path) or not is_path_existent(
+            self.db_path
+        ):
+            self.conn = sqlite3.connect(self.db_path)
+            self.cursor = self.conn.cursor()
+
+    def _create_database_if_not_exist(self) -> None:
+        """
+        Creates the database schema if necessary.
+
+        If the necessary tables do not exist in the database,
+            it creates the database schema from the schema file.
+        """
+        if not self._check_tables():
+            self._create_database(self.schema_file)
+
+    def _create_database(self, schema_file: str) -> None:
+        """
+        Creates the database schema if necessary.
+
+        Args:
+            schema_file (str): The path to the schema file containing
+                the database schema definition.
+        """
+        self._initialize_database_from_schema(schema_file)
+
+    def _check_tables(self) -> None:
```

Comment:
if_table_exists

is more meaningful than "check_tables"

Reply:
This goes against the practice of having verbs as function names